### PR TITLE
bug - Component "error" length issue

### PR DESCRIPTION
**Pull request by PipoCanaja**
This DB field is limited to 255 chars currently. This PR checks for longer strings to avoid errors.

```
Next Illuminate\Database\QueryException: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'error' at row 1 (Connection: mysql, SQL: update `component` set `error` = Pool Member is Down: /Common/monitor-xxxxxxxxxx-externalws.xxxxxxxxxx.xxxxxxxxxx-reverse_proxy_front_end_https: No successful responses received before deadline. @2024/03/25 13:53:15. /Common/monitor-xxxxxxxxxx-externalws.xxxxxxxxxx.xxxxxxxxxx-reverse_proxy_tcp_5061: No successful responses received before deadline. @2024/03/25 13:53:15. where `id` = 11745) in /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(783): Illuminate\Database\Connection->runQueryCallback('update `compone...', Array, Object(Closure))
#1 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(619): Illuminate\Database\Connection->run('update `compone...', Array, Object(Closure))
#2 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(552): Illuminate\Database\Connection->affectingStatement('update `compone...', Array)
#3 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3487): Illuminate\Database\Connection->update('update `compone...', Array)
#4 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(1060): Illuminate\Database\Query\Builder->update(Array)
#5 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1213): Illuminate\Database\Eloquent\Builder->update(Array)
#6 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1130): Illuminate\Database\Eloquent\Model->performUpdate(Object(Illuminate\Database\Eloquent\Builder))
#7 /opt/librenms/LibreNMS/Component.php(244): Illuminate\Database\Eloquent\Model->save()
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php(240): LibreNMS\Component->LibreNMS\{closure}(Object(App\Models\Component), 453)
#9 /opt/librenms/LibreNMS/Component.php(275): Illuminate\Support\Collection->each(Object(Closure))
#10 /opt/librenms/includes/polling/loadbalancers/f5-ltm.inc.php(214): LibreNMS\Component->setComponentPrefs(566, Array)
#11 /opt/librenms/includes/polling/loadbalancers.inc.php(13): include('/opt/librenms/i...')
#12 /opt/librenms/LibreNMS/Modules/LegacyModule.php(114): include('/opt/librenms/i...')
#13 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\LegacyModule->poll(Object(LibreNMS\OS\F5), Object(LibreNMS\Data\Store\Datastore))
#14 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#15 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#16 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle(Object(App\Polling\Measure\MeasurementManager))
#17 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
```
DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Shouldnt the field be extended instead?

File: LibreNMS/Component.php [hunk 1]
```diff
@@ -229,6 +229,15 @@ public function setComponentPrefs($device_id, $updated)
 
                 // update component attributes
                 $component->fill($update);
+                // sanity check
+                if (! is_null($component->error)) {
```
Comment (murrant) [inline LibreNMS/Component.php:233]: might be better to put these in the model as mutators?

Comment (PipoCanaja) [reply LibreNMS/Component.php:233]: Good occasion to learn how a mutator works :) just had a look at the doc and it makes sense indeed. 
Will send a patch.

Comment (PipoCanaja) [reply LibreNMS/Component.php]: Should be fixed now. But code is currently not tested.

Comment (PipoCanaja) [reply LibreNMS/Component.php]: Hi @murrant 
Seems that my mutators do not get called. I still receive the Exceptions if I disable the test in the LTM poller code. 
```
Error polling loadbalancers module for xxxxxxxxxxx. PDOException: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'error' at row 1 in /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php:612
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(612): PDOStatement->execute()
#1 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(816): Illuminate\Database\Connection->Illuminate\Database\{closure}('update `compone...', Array)
#2 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(783): Illuminate\Database\Connection->runQueryCallback('update `compone...', Array, Object(Closure))
#3 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(619): Illuminate\Database\Connection->run('update `compone...', Array, Object(Closure))
#4 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Connection.php(552): Illuminate\Database\Connection->affectingStatement('update `compone...', Array)
#5 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3487): Illuminate\Database\Connection->update('update `compone...', Array)
#6 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(1060): Illuminate\Database\Query\Builder->update(Array)
#7 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1213): Illuminate\Database\Eloquent\Builder->update(Array)
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1130): Illuminate\Database\Eloquent\Model->performUpdate(Object(Illuminate\Database\Eloquent\Builder))
#9 /opt/librenms/LibreNMS/Component.php(270): Illuminate\Database\Eloquent\Model->save()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php(240): LibreNMS\Component->LibreNMS\{closure}(Object(App\Models\Component), 453)
#11 /opt/librenms/LibreNMS/Component.php(301): Illuminate\Support\Collection->each(Object(Closure))
#12 /opt/librenms/includes/polling/loadbalancers/f5-ltm.inc.php(214): LibreNMS\Component->setComponentPrefs(566, Array)
#13 /opt/librenms/includes/polling/loadbalancers.inc.php(13): include('/opt/librenms/i...')
#14 /opt/librenms/LibreNMS/Modules/LegacyModule.php(114): include('/opt/librenms/i...')
#15 /opt/librenms/LibreNMS/Poller.php(176): LibreNMS\Modules\LegacyModule->poll(Object(LibreNMS\OS\F5), Object(LibreNMS\Data\Store\Datastore))
#16 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#17 /opt/librenms/app/Console/Commands/DevicePoll.php(45): LibreNMS\Poller->poll()
#18 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle(Object(App\Polling\Measure\MeasurementManager))
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#20 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#21 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#22 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#23 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call(Array)
#24 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#25 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(181): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#26 /opt/librenms/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /opt/librenms/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand(Object(App\Console\Commands\DevicePoll), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /opt/librenms/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 /opt/librenms/lnms(40): App\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#32 {main}
```

**Review by murrant — CHANGES_REQUESTED** [review]
Hmm, probably should use the regular model in app/Models/Component.php instead of updating the compatibility shim.

**Comment by PipoCanaja** [comment]
> Hmm, probably should use the regular model in app/Models/Component.php instead of updating the compatibility shim.

I re-read this sentence a few times, but I don't get it :) Do you have a "Laravel for dummies" kind of clue for me ? :D

**Comment by murrant** [comment]
LibreNMS/Component.php is just there to allow old code to use app/Models/Component.php without being updated.

**Comment by PipoCanaja** [comment]
Hi @murrant 
Seems that this time it works as expected. I removed the checks on the F5 code, and we leave it to the Mutators to crop the data before write. What do you think ?

**Comment by PipoCanaja** [comment]
@murrant I am a little bit overloaded with messages. What I can tell is that this code seems to correctly protect from the original issue : error being too long to be stored in components. 
But I probably don't know how to write that in a safe and bulletproof way :)

**Comment by PipoCanaja** [comment]
Seems that the typing issues are now fixed. Testing this in my production for a few days

**Comment by PipoCanaja** [comment]
@librenms/reviewers All good to me after a few days in my prod. Ready for review.

**Comment by PipoCanaja** [comment]
@librenms/reviewers It is now good to merge. No issue on my side.

**Review by murrant — APPROVED** [review]

**Pull request merged**